src/components/ProgressIndicator/_progress-indicator.scss

```diff
@@ -1,82 +1,306 @@
+    top: 11px;
+    left: 24px;
+
+    &-complete {
+      @extend .line;
+      background-color: #0f62fe;
     }
 
-    .#{$prefix}--progress-label {
-      max-width: calc(100% - 2.2rem);
-      position: relative;
+    &-sub {
+      @extend .line;
+      left: 16px;
+      width: calc(100% - 16px);
+
+      &-complete {
+        @extend .line-sub;
+        background-color: #0f62fe;
+      }
     }
+  }
 
-    .#{$prefix}--progress-optional {
-      margin-left: $spacing-08;
+  .step-button {
+    display: flex;
+    position: relative;
+    text-align: left;
+    width: 8rem;
+    height: 68px;
+    padding: 0;
+    margin: 0;
+
+    background-color: Transparent;
+    background-repeat: no-repeat;
+    cursor: default;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    outline: none;
+    border: none;
+
+    &.clickable {
+      .label {
+        cursor: pointer;
+      }
+    }
+
+    &:focus {
+      .label:not(.label-optional) {
+        box-shadow: 0 rem(3px) 0 0 $link-01;
+      }
+    }
+
+    &:hover {
+      .label:not(.label-optional) {
+        box-shadow: 0 rem(1px) $link-01;
+      }
+    }
+
+    &:active {
+      .label:not(.label-optional) {
+        box-shadow: 0 rem(3px) 0 0 $interactive-01;
+      }
     }
   }
 
-  .#{$prefix}--progress-optional {
-    margin-left: $spacing-07;
+  .hidden {
+    visibility: hidden;
   }
 
-  .#{$prefix}--progress-step svg {
-    margin-left: $spacing-03;
-    margin-right: $spacing-03;
+  .sub-step {
+    .icon {
+      margin-top: 4px;
+      margin-left: 0;
+    }
   }
 
-  .#{$prefix}--progress-line {
-    top: 0;
+  svg text {
+    font-size: 18px;
   }
-}
 
-.#{$prefix}--progress-line {
-  width: 100%;
-}
+  //-------States
+
+  .step-current {
+    &:not(.step-invalid) {
+      .main-step {
+        fill: #0f62fe;
+
+        text {
+          fill: white;
+        }
+      }
 
-.#{$prefix}--skeleton {
-  .#{$prefix}--progress-step {
-    flex: unset;
+      .sub-step {
+        svg {
+          fill: #0f62fe;
+        }
+      }
+    }
   }
-}
 
-html[dir='rtl'] {
-  .#{$iot-prefix}--progress-indicator {
-    &.#{$prefix}--progress--vertical {
-      .#{$prefix}--progress-step-button {
-        text-align: right;
+  .step-complete {
+    svg {
+      fill: #0f62fe;
+    }
+  }
+
+  .step-incomplete {
+    .main-step {
+      fill: #171717;
+    }
+
+    .sub-step {
+      svg {
+        color: #171717;
+      }
+    }
+  }
+
+  .step-disabled {
+    &:not(.step-invalid) {
+      .main-step {
+        fill: #bebebe;
       }
-      .#{$prefix}--progress-optional {
-        margin-right: $spacing-08;
+
+      .sub-step {
+        svg {
+          fill: #bebebe;
+        }
       }
     }
 
-    .#{$prefix}--progress-optional {
-      margin-right: $spacing-07;
-      right: 0;
-      margin-left: 0;
+    &.step-invalid {
+      svg {
+        fill: #bebebe;
+      }
+    }
+
+    .label {
+      @extend .label;
+      color: #c6c6c6;
+      cursor: default !important;
```

Comment:
we should not have `!important`. If higher specificity is needed it should be gradually raised through higher specificity .